lib/Service/SocialApiService.php

```diff
@@ -217,8 +216,51 @@ public function updateContact(string $addressbookId, string $contactId, string $
 	}
 
 	/**
-	 * @NoAdminRequired
+	 * checks an addressbook is existing
+	 *
+	 * @param {string} searchBookId the UID of the addressbook to verify
+	 * @param {string} userId the user that should have access
+	 *
+	 * @returns {bool} true if the addressbook exists
+	 */
+	public function existsAddressBook($searchBookId, $userId) {
+		$manager = $this->manager;
+		$coma = new ContactsManager($this->davBackend, $this->l10n);
+		$coma->setupContactsProvider($manager, $userId, $this->urlGen);
+		$addressBooks = $manager->getUserAddressBooks();
+		if ($this->getAddressBook($searchBookId, $manager) == null) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * checks a contact exists in an addressbook
+	 *
+	 * @param {string} searchContactId the UID of the contact to verify
+	 * @param {string} searchBookId the UID of the addressbook to look in
+	 * @param {string} userId the user that should have access
 	 *
+	 * @returns {bool} true if the contact exists
```

Comment:
```suggestion
	 * @returns bool true if the contact exists
```

We don't use `{` in php for those, but it seems to be an overall problem in this app.